### ISSUE TITLE
Egg orientation seems to be reverted

### ISSUE THREAD

**Issue by illwieckz** [Unvanquished/Unvanquished#1356]
When you build alien models in Unvanquished, they're usually facing at you.

When you build egg, the egg does not have eyes so the egg itself does not really have orientation, but aliens spawning from the egg will be facing at you.

But when you edit maps in NetRadiant, you have to turn the egg 180° on the vertical axis to get the player spawning with the desired orientation.

Here is a screenshot of alien buildables as seen in NetRadiant. All those models were configured to use a `0` vertical angle, so, they are expected to all face in the same direction and look at you. Though, if you do that, the alien players will spawn showing you their back, on the opposite orientation than buildables.

[![model orientation](https://dl.illwieckz.net/b/unvanquished/bugs/model-orientation/20210329-025136-000.netradiant-unvanquished-alien-model-orientation.webp)](https://dl.illwieckz.net/b/unvanquished/bugs/model-orientation/20210329-025136-000.netradiant-unvanquished-alien-model-orientation.webp)

I painted some stuff on the model editor preview texture to make it obvious. The green line is the “face” of the egg, the red arrow tells the direction of the spawn. So, when doing a map with this position, a mapper has to orientate all the model with the `0` vertical angle, while having to set `180` for the egg to get alien players spawn correctly. Otherwise, if you set the egg close to a wall and keep the original orientation, the player just hit the wall while spawning.

**Comment by bmorel** [comment]
I think this issue should be in netradiant, not here.

**Comment by illwieckz** [comment]
No, it's an issue in our gamecode. I illustrate with NetRadiant screenshot because mappers have to turn the model on 180° way to have the game spawns the player the expected way.

**Comment by bmorel** [comment]
Are not aliens just thrown away by the egg anyway?

**Comment by illwieckz** [comment]
They're thrown away in a precise direction.

**Comment by bmorel** [comment]
Eggs don't have eyes... but anyway, in the following (in-game) screenshot, both buildings were built with same direction:
![2022-07-16-023025_1362x748_scrot](https://user-images.githubusercontent.com/1316300/179327137-eda94b6b-f44d-43b2-b092-8f70ab4ee945.png)

When I spawn, I face the same direction as the barricade. To me, this seems to correct. So maybe it was fixed since then.

**Comment by illwieckz** [comment]
It is not. If eggs had eyes and a butt, it would show you its butt. But spawning aliens would show you their eyes. That's why the egg is said to be reverted. Mappers have to put it wrong to make sure player spawn correctly.

One issue is that by fixing this, every legacy layouts in legacy maps will spawn aliens in the wrong direction. So, if someone fixed it without anyway to handle legacy maps (if there is a way), _you would know about it_.

**Comment by bmorel** [comment]
I've built the barricade and the egg by just straffing. Respawning made me looking in the same direction as the barricade is looking, which seems a normal behavior to me.
The problem of legacy, unmaintained maps, is that they're unmaintained. This is a different problem, especially since many are non-free or have lost source, which is almost the same.
Imo only official maps should be considered for update when a bug is fixed.

I still disagree there's a bug in the current nightly build anyway. When I spawned, I was looking the inverted direction I was looking at when building, which is consistent with all other buildings, even if I consider this behavior dumb, since it implies human builders notably have to expose their back when they're trying to build turrets placed correctly to defend a place.

**Comment by illwieckz** [comment]
Have your read my previous message**s** ?

The egg is 180° wrong relatively to other buildables. But the spawning alien is 180° wrong relatively to the egg, that's why the spawning alien is right.

But since the egg is wrong (and the spawning is wrong), mappers have to 180° turn eggs when mapping.

In game you don't experience the issue, but as a mapper you have to turn every egg the wrong way. But this is a bug in the game.

As a mapper you have to turn every egg the wrong way to workaround the game bug.

**Comment by bmorel** [comment]
> Have your read my previous messages ?

I did.

> The egg is 180° wrong relatively to other buildables. But the spawning alien is 180° wrong relatively to the egg, that's why the spawning alien is right.

Ok, so the alien's spawn have 2 bugs which cancels themselves?

**Comment by illwieckz** [comment]
Yes, for gamers the 2 bugs cancels themselves.

But mappers experience the first bug because they have to implement the first bug on purpose to workaround the game bug.

**Comment by bmorel** [comment]
Ok. Now it's clear, so I tagged the issue accordingly.

**Comment by illwieckz** [comment]
As an additional comment, in my original screenshot, the arrow on the egg gives the orientation of the spawning alien: the spawning alien will face the overmind (spawning alien eyes facing overmind eye).

The green line on the egg gives the real orientation of the egg, where eyes would be if eggs had eyes, where spawning alien eyes have to be (but are not).

[![model orientation](https://dl.illwieckz.net/b/unvanquished/bugs/model-orientation/20210329-025136-000.netradiant-unvanquished-alien-model-orientation.webp)](https://dl.illwieckz.net/b/unvanquished/bugs/model-orientation/20210329-025136-000.netradiant-unvanquished-alien-model-orientation.webp)

In game, if you have this view and place an egg close to the wall (where the OM) is, the alien will spawn facing you. In the level editor, if you do exactly the same, the alien will spawn facing the wall.

**Comment by slipher** [comment]
There is a discussion of why this happens at https://github.com/Unvanquished/Unvanquished/pull/3439. (The code just arbitrarily adds 180 degrees to an angle for the egg and telenode models.)

**Comment by slipher** [comment]
I imagine that Radiants don't read our `.model.cfg` files, so what if we just rotate the models 180 degrees in source, then cancel that out with our `.model.cfg` files? That should fix the Radiant issue, without having to define multiple versions of the entity data interface.

**Comment by slipher** [comment]
Never mind, reading that more closely, the issue relates to the orientation of the models relative to the view position in Radiant, not the visual appearance of the models. So my suggestion is pointless.

**Comment by slipher** [comment]
I guess what you could do is define a different class name. Like `team_alien_spawn` spawns an egg with the compatibility random 180 degree rotation, but `team_alien_egg` spawns an egg facing the direction it ought to.
